### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by DerCalli**
Description

Refactor as discussed

- no more additional plugins for products inside datalayer
- add new dependency to spryker/money
- remove old plugins, replaced by `CategoryDataLayerExpanderPluginExpanderPlugin`
- rename model to expander
- additional expander for category-products
- adjust unit-tests

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -10,6 +10,7 @@
   ],
   "require": {
     "php": ">=7.2",
+    "spryker/money": "^2.9",
```
Comment (daniel-rose) [inline composer.json]: Please add all possible versions here...

File: src/FondOfSpryker/Shared/GoogleTagManagerCategoryConnector/Transfer/google_tag_manager_category_connector.transfer.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0"?>
```
Comment (daniel-rose) [inline src/FondOfSpryker/Shared/GoogleTagManagerCategoryConnector/Transfer/google_tag_manager_category_connector.transfer.xml]: Unused stuff in this package...

File: src/FondOfSpryker/Yves/GoogleTagManagerCategoryConnector/Expander/CategoryDataLayerExpander.php [hunk 3]
```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace FondOfSpryker\Yves\GoogleTagManagerCategoryConnector\Expander;
+
+use FondOfSpryker\Shared\GoogleTagManagerCategoryConnector\GoogleTagManagerCategoryConstants as ModuleConstants;
+
+class CategoryDataLayerExpander implements CategoryDataLayerExpanderInterface
```
Comment (daniel-rose) [inline src/FondOfSpryker/Yves/GoogleTagManagerCategoryConnector/Expander/CategoryDataLayerExpander.php]: Use this pattern for your expander in this module: https://refactoring.guru/design-patterns/chain-of-responsibility/php/example

**Review by daniel-rose — CHANGES_REQUESTED** [review]

File: src/FondOfSpryker/Yves/GoogleTagManagerCategoryConnector/Expander/DataLayerExpander.php [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+     * @var \FondOfSpryker\Yves\GoogleTagManagerCategoryConnector\Expander\ProductDataLayerExpanderInterface
+     */
+    protected $productExpander;
+
+    /**
+     * @param \FondOfSpryker\Yves\GoogleTagManagerCategoryConnector\Expander\ProductDataLayerExpanderInterface $productExpander
+     */
+    public function __construct(ProductDataLayerExpanderInterface $productExpander)
+    {
+        $this->productExpander = $productExpander;
```
Comment (daniel-rose) [inline src/FondOfSpryker/Yves/GoogleTagManagerCategoryConnector/Expander/DataLayerExpander.php:19]: Make this sense? I think the `productExpander` must be an own plugin in the module `fond-of-spryker/google-tag-manager-product-connector`...

**Review by daniel-rose — CHANGES_REQUESTED** [review]

File: src/FondOfSpryker/Yves/GoogleTagManagerCategoryConnector/Expander/DataLayerExpander.php (hunk 4 above)
Comment (DerCalli) [reply src/FondOfSpryker/Yves/GoogleTagManagerCategoryConnector/Expander/DataLayerExpander.php:19]: Mhhhh, not sure about that. we already talked about that and we said that we don't want to do it that way

- dependency between packages
- need to overwrite module dependency provider in pyz
- need to handle different logic based on source for product-propertie (product-detail, category, quote etc.)

**Pull request merged**